civis/futures.py

```diff
@@ -228,7 +228,12 @@ def _exception_from_logs(self, exc, nlog=15):
             exc = MemoryError(mem_err[0])
         else:
             # Unknown error; return logs to the user as a sort of traceback
-            all_logs = '\n'.join([x['message'] for x in logs])
+            job_run_ids_in_exc = _format_job_run_ids_in_exception(
+                self.job_id, self.run_id)
+            all_logs = (
+                f"{job_run_ids_in_exc} "
+                + '\n'.join([x['message'] for x in logs])
+            )
```

Comment:
These changes as-is are in `ContainerFuture` instead of the parent class `CivisFuture`. Because it seems to be much more common for users to write code like `fut = CivisFuture(...)` rather than using `ContainerFuture`, I'm wondering whether it's worth doing the following:

1. Move these changes from `ContainerFuture` to `CivisFuture`
2. Move the changes from #416 from `ContainerFuture` to `CivisFuture`

Hi @ggarcia-civis , I have questions for you:

* What do you think about 1 and 2 here? Are both or one of them worth the extra work? Or are there actually potential blockers? Speaking of blockers, my next question below...
* Do you recall why #416 was done in `ContainerFuture` but not `CivisFuture`?

Reply:
On the ticket Eli mentioned doing it on `ContainerFuture` instead of `CivisFuture` because there were some instances of `CivisFuture` that already had error handling and not wanting to mess with that.